cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -1108,7 +1117,7 @@ def create_rbac_resources():
     # NB: when metrics and logs are retrieved by proxy, the 'user' is the
     # common name of the cert used to authenticate the proxied request.
     # The CN for /root/cdk/client.crt is 'client'.
-    proxy_user = 'client'
+    proxy_user = 'system:kube-apiserver'
```

Comment:
:+1:

Reply:
@johnsca is `system:kube-apiserver` the CN of the cert now, or is that just the role that you expect proxied requests to use? If the former, great, update the comment 2 lines above...

If the latter, i have a concern.  When external tools (like prometheus) try to access metrics or logs, they will do so via the apiserver proxy, which afaik makes a request to the actual worker as the CN of the certificate used between the k8s-master and workers:

https://kubernetes.io/docs/reference/access-authn-authz/authentication/#x509-client-certs

_...the common name of the subject is used as the user name for the request..._

When RBAC is enabled, I'm pretty sure we'll need the `proxy_user` here set to the CN of the cert to allow these types of requests.

Reply:
It's the former. The CN is now `system:kube-apiserver` on kubernetes-master units. You can see the client cert being requested with that CN [here](https://github.com/juju-solutions/kubernetes/pull/207/files#diff-e6f5b987d0e976d7b76acd341abc74c0R835).

Your point about the comment is valid however.